versioned/mongodb/src/main/java/com/dremio/nessie/versioned/store/mongodb/MongoDBStore.java

```diff
@@ -228,21 +244,30 @@ public void load(LoadStep loadstep) throws NotFoundException {
     Preconditions.checkArgument(type.getObjectClass().isAssignableFrom(value.getClass()),
         "ValueType %s doesn't extend expected type %s.", value.getClass().getName(), type.getObjectClass().getName());
 
-    // TODO: Handle ConditionExpressions.
-    if (conditionUnAliased.isPresent()) {
-      throw new UnsupportedOperationException("ConditionExpressions are not supported with MongoDB yet.");
-    }
-
     final MongoCollection<V> collection = getCollection(type);
+    final Bson idFilter = Filters.eq(Store.KEY_NAME, ((HasId) value).getId());
+    if (conditionUnAliased.isPresent()) {
+      final Bson fullFilter = Filters.and(idFilter, conditionUnAliased.get().accept(CONDITION_VISITOR));
 
-    // Use upsert so that if an item does not exist, it will be insert.
-    Mono.from(collection.replaceOne(Filters.eq(Store.KEY_NAME, ((HasId)value).getId()), value, new ReplaceOptions().upsert(true)))
-        .block(timeout);
+      if (1 != Mono.from(collection.replaceOne(fullFilter, value)).block(timeout).getModifiedCount()) {
+        throw new ConditionFailedException("Condition failed during put operation.");
+      }
+    } else {
+      // Use upsert so that if an item does not exist, it will be inserted.
+      Mono.from(collection.replaceOne(idFilter, value, new ReplaceOptions().upsert(true))).block(timeout);
+    }
   }
 
   @Override
   public boolean delete(ValueType type, Id id, Optional<ConditionExpression> condition) {
-    throw new UnsupportedOperationException();
+    final MongoCollection<?> collection = getCollection(type);
+
+    Bson filter = Filters.eq(Store.KEY_NAME, id.getId());
```

Comment:
Something similar was called idFilter in the previous function? Are they similar or very different?
If similar, let's name it idFilter here too, so developers know it's the same thing.